docs/getting-started.md

```diff
@@ -0,0 +1,439 @@
+
+The state of the container will not be saved when the shell prompt exits. Best practice is modify the Dockerfile to modify the image.
+
+**Testing in a Container Workflow**
+
+1. Edit the changes to OSTree on your local machine.
+2. `git add` to stage the changed files, `git commit` and then `git push origin <local-branch>:<remote-branch>`.
+3. Testing on a _new_ container vs. Testing on an _existing_ container:
+
+    If the `ostree-testing` container was newly built right after your changes have been committed, then the container's build of `ostree` should contain your changes.
+
+    Else: Within the `ostree-testing` container, run `../ostree-test.sh` in the ostree directory. This will pull in changes from your branch and create a new `ostree` build.
+
+4. `make install` will install OSTree in the default location i.e. `/usr/local/..`in a Fedora 28 container.  
+
+5. Test `ostree`.
+
+### Testing in a Virtual Machine
+
+See this guide on creating an [Atomic Host Vagrant VM](https://gist.github.com/Bubblemelon/bd9f3cf429d9e04be52f9987baed53c8).
+
+You may want to use `scp` or `rsync` to transfer `ostree` files to a Vagrant VM.
+
+To find the IP address of a Vagrant VM, run `vagrant ssh-config` in the same directory as the `Vagrantfile`.
+
+**Steps to `scp` specific `ostree` files to test a build**:
+
+1. From any directory on the terminal, run `scp -r </usr/bin/ostree> </usr/lib64/libostree-1.so.1> vagrant@<ip-address>:~`
+
+2. Run `vagrant ssh` to login to the VM.
+
+3. To override the `Read-only file system` warning, run `ostree admin unlock`
+
+4. Now replace the original OSTree binary with the new one by running `sudo mv ~/ostree /usr/bin/ostree` and as well as the shared object file by running `sudo mv ~/libostree-1.so.1 usr/lib64/libostree-1.so.1`
+
+If the `mv` command was successful, running `which ostree` shouldn't return an error. Proceed to test the new OSTree binary, by running the OSTree commands that you've changed.
+
+**Steps to `rsync` files to test an `ostree` build**:
+
+1. Copy your public ssh key from your host machine e.g. `id_rsa.pub` to `/home/vagrant/.ssh/authorized_keys` on the VM.
+
+2. Run `sudo su`, followed by `ssh localhost` then press <kbd>Ctrl</kbd>+<kbd>c</kbd> to exit from the decision prompt. This will create the `.ssh` file with the right permissions.
+
+3. Using `Vagrant` as the user, run `sudo cp ~/.ssh/authorized_keys /root/.ssh/`. So that user `root` has the the same login credentials.
+
+4. `<ostree-install-dir>` will serve as the local install location for `ostree` and the path to this directory should be _absolute_ when specified in `DESTDIR`.
+
+5. Run the following commands to build OSTree for Atomic Host on your host machine:
+
+        ./autogen.sh
+        ./configure --prefix=/usr
```

Comment:
First, you *absolutely need* to match `--libdir` which on Fedora is `/usr/lib64` for 64 bit hosts.  It's different for Debian (this is a tricky aspect of all of this...it's a bit sad there's not a portable "configure like the host system" wrapper).  I think meson solves this by having the right defaults.

It's also pretty important to specify `--sysconfdir=/etc`.

It's probably simpler to just recommend `./autogen.sh --prefix=/usr --libdir=/usr/lib64 --sysconfdir=/etc`; it'll be wrong for Debian but oh well.

Reply:
Hmm.. funny, I've built ostree master using those commands and it seems to work fine. Also tested some of ostree's commands. Could you give me an example of what might not work, when built without those flags?

Reply:
If you change code that ends up in the CLI (`/usr/bin/ostree`) as this guide shows it'll work.  If you change code that ends up in the shared library (e.g. `src/libostree/ostree-repo.c`), your changes won't appear because you'll be creating a new `/usr/lib/libostree-1.so.1.0.0` instead of overwriting the `/usr/lib64/libostree-1.so.1.0.0` .

Reply:
Aside: You might wonder "what's up with this libdir stuff, why isn't it /usr/lib"?  It's basically that Fedora chose one way many years ago (`lib64`) which is known as "multilib" (solving the 32 and 64 bit parallel install), but Debian took a few more years and implemented a [more general system called multiarch](https://wiki.debian.org/Multiarch/HOWTO).